nodejs/README.adoc

```diff
@@ -0,0 +1,341 @@
+endif::[]
+
+[id="proc-running-nodejs-example-application_{context}"]
+== Running the Node.js example application
+
+After you configure your Node.js application to connect to a Kafka instance and you create the required Kafka topic, you can run the Node.js application to start producing and consuming messages.
+
+In this task, you'll use the following components of the Node.js application to produce and consume messages:
+
+* A `producer-backend` component that generates random country names and sends these names to the Kafka topic.
+* A `consumer-backend` component that consumes the country names from the Kafka topic.
+
+.Prerequisites
+* You've configured the Node.js example application to connect to the Kafka instance.
+* You've created the `countries` example Kafka topic.
+
+.Procedure
+. On the command line, navigate to the `reactive-example` directory corresponding to the repository that you cloned.
++
+.Navigating to the reactive-example directory
+[source]
+----
+$ cd ~/reactive-example
+----
+
+. Check out the `kafkajs` branch of the repository.
++
+.Checking out the kafkajs branch
+[source]
+----
+$ git checkout kafkajs
+----
+
+. Run the shell script that sets environment variables for the bootstrap server endpoint and client credentials. These are required to connect to your Kafka instance.
++
+.Running the shell script
+[source]
+----
+$ source set-environment.sh
+----
+
+. Navigate to the directory for the consumer component. Use Node Package Manager (npm) to install the package for this component.
++
+.Installing the consumer component
+[source]
+----
+$ cd consumer-backend
+$ npm install
+----
+
+. Run the consumer component.
++
+.Running the consumer component
+[source]
+----
+$ node consumer.js
+----
++
+You should see the Node.js application start to run and connect to the Kafka instance. However, because you haven't run the producer component yet, the consumer has no country names to display.
++
+If the application fails to run, review the error log in the command-line window and address any problems. Also, review the steps in this quick start to ensure that the application and Kafka topic are configured correctly.
+
+. On your computer, open a second command-line window or tab. On the command line, navigate to the `reactive-example` directory corresponding to the repository that you cloned.
++
+.Navigating to the reactive-example directory
+[source]
+----
+$ cd ~/reactive-example
+----
+
+. Check out the `kafkajs` branch of the repository.
++
+.Checking out the kafkajs branch
+[source]
+----
+$ git checkout kafkajs
+----
+
+. Run the shell script that sets environment variables for the bootstrap server endpoint and client credentials.
++
+.Running the shell script
+[source]
+----
+$ source set-environment.sh
+----
+
+. Navigate to the directory for the producer component. Use Node Package Manager (npm) to install the package for this component.
++
+.Installing the producer component
+[source]
+----
+$ cd producer-backend
```

Comment:
While those steps are easy it felt that we could have them grouped together with some script on the repository.
For example npm workspace can run multiple projects at the same time with the proper logs.

Reply:
I sometimes find that when its just run this script then that script you don't see what is going on. Since these are quite simple I"d lean towards leaving the individual steps. They don't take long and you see the exact steps.

Reply:
Agreed.